mem/vm/mmu/mmu.go

```diff
@@ -95,7 +101,14 @@ func (m *middleware) finalizePageWalk(
 	page, found := m.pageTable.Find(req.PID, req.VAddr)
 
 	if !found {
-		panic("page not found")
+		if m.autoPageAllocation {
+			page = m.createDefaultPage(req.PID, req.VAddr, req.DeviceID)
+			m.pageTable.Insert(page)
+		} else {
+			panic("page not found")
+		}
+	} else if m.autoPageAllocation {
```

Comment:
This not gonna work. What if the page has never been walked before but we want to make the allocation? 

Let's do this. We can add a `ReverseLookup` method in PageTable (probably need new data structure to prevent traversing). Then in `allocatePhysicalPage` method, for every candidate page, we do a reverse lookup. If the physical page is allocated, we try next page.